src/cp/cp.rs

```diff
@@ -579,7 +579,7 @@ impl Options {
         let recursive = matches.is_present(OPT_RECURSIVE) || matches.is_present(OPT_RECURSIVE_ALIAS)
             || matches.is_present(OPT_ARCHIVE);
 
-        let backup = matches.is_present(OPT_BACKUP) || matches.is_present(OPT_SUFFIX);
+        let backup = matches.is_present(OPT_BACKUP) || (matches.occurrences_of(OPT_SUFFIX) > 0);
```

Comment:
IIUC this is required because `OPT_SUFFIX` has a default value, right?

Reply:
Yes, because of the default value `OPT_SUFFIX` always has a value even if it wasn't supplied on the command line. Using `occurrences_of(OPT_SUFFIX)` tests whether it was actually supplied as a command line argument which stops `cp` from always creating a backup, even when not requested.